Frontend-Backend Connect v0.1 :D

Xapollon430 opened the pull request:
Chloe edit: Connected basic translation API backend to landing page frontend! Frontend now displays a single translation result (or none if no translation is available) after a user enters their translation information and presses submit. (Will later be modified to return multiple results if possible.) 

![image](https://user-images.githubusercontent.com/53862535/117512219-3753f300-af5d-11eb-9da3-be824ea8a9e7.png)

Xapollon430 committed:
landing work

Xapollon430 committed:
landing work

Xapollon430 committed:
landing work

Xapollon430 committed:
modal

Xapollon430 committed:
report modal

Xapollon430 committed:
react router fix

Xapollon430 committed:
header

Xapollon430 committed:
Merge branch 'main' of https://github.com/candreso/CodeLingual

Xapollon430 committed:
Merge branch 'main' of https://github.com/candreso/CodeLingual

Xapollon430 committed:
pr

candreso committed:
removed repeated code in routes.js

candreso committed:
restored server.js version from candreso/CodeLingual and registered * route to react pages

Xapollon430 committed:
fixed route order

Xapollon430 committed:
es6 tricks

Xapollon430 committed:
pre-pull

Xapollon430 committed:
pre-heroku

Xapollon430 committed:
frontend work

Xapollon430 committed:
tying code together

Xapollon430 committed:
Apply linting and formatting changes

Xapollon430 commented:
:D

Xapollon430 commented:
:D

candreso commented on client/public/index.html:
```diff
@@ -16,4 +16,5 @@
     <div id="backdrop-hook"></div>
     <div id="root"></div>
   </body>
+ 
```
To be removed in future PR

candreso committed:
minor clean-up

candreso committed:
reverted destructuring one-liner back to simple assignments for increased readability

candreso committed:
updated some dependencies to fix vulnerabilities

candreso committed:
Fixed client production build error

- error caused all static chunks to solely consist of index.html
  contents
- fixed by explicitly telling the server to use the build files when the
  "/" (home) endpoint is hit

candreso committed:
didn't need to configure the env again

candreso commented on client/src/components/Landing/Landing.js:
```diff
@@ -33,9 +34,16 @@ const Landing = () => {
+        setTranslations(data);
+      }
+
+      history.push(ENDPOINT);
```
later I want to see if I can get the new url to just be "localhost:PORT?sl=..." instead of what it is now 👇 
![image](https://user-images.githubusercontent.com/53862535/117512317-610d1a00-af5d-11eb-9176-bdc8ef5b1bad.png)

candreso commented on client/src/components/Landing/Landing.js:
```diff
@@ -33,9 +34,16 @@ const Landing = () => {
+      const ENDPOINT = `?sl=${formData.sl}&dl=${formData.dl}&code=${formData.code}`;
 
-      history.push("?sl=c&dl=javascript&code=console.log()");
+      const results = await fetch(`http://localhost:5000/api/v1${ENDPOINT}`);
```
in the near future I also need to figure out how to refactor the server port so that the server and client apps are retrieving 1 variable value from the same location

candreso commented on client/src/components/TranslationsBox/TranslationsBox.js:
```diff
@@ -12,14 +12,14 @@ import { Button } from "@material-ui/core";
-  "export default colorPicker",
-  "import lodash from {lodash}",
-];
+// const fakeData = [
+//   "System.out.println('123')",
+//   "x.toString().toHash()",
+//   "export default colorPicker",
+//   "import lodash from {lodash}",
+// ];
```
should probs remove at some point

candreso commented on client/src/components/TranslationsBox/TranslationsBox.js:
```diff
@@ -41,73 +41,73 @@ const TranslationsBox = ({ formData }) => {
-          </div>
-        );
-      })}
+      {/* })} */}
```
@Xapollon430 can I delete this in a future PR?

candreso reviewed (approved):
:o)
